clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/provider/view/EcrImageProvider.java

```diff
@@ -0,0 +1,188 @@
+import com.amazonaws.services.ecr.model.ListImagesRequest;
+import com.amazonaws.services.ecr.model.ListImagesResult;
+import com.netflix.spinnaker.clouddriver.aws.security.AmazonClientProvider;
+import com.netflix.spinnaker.clouddriver.aws.security.AmazonCredentials;
+import com.netflix.spinnaker.clouddriver.aws.security.NetflixAmazonCredentials;
+import com.netflix.spinnaker.clouddriver.ecs.model.EcsDockerImage;
+import com.netflix.spinnaker.clouddriver.security.AccountCredentials;
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsProvider;
+import com.netflix.spinnaker.kork.web.exceptions.NotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.util.Collections;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+@Component
+public class EcrImageProvider implements ImageRepositoryProvider {
+  private static final Pattern ACCOUNT_ID_PATTERN = Pattern.compile("^([0-9]{12})");
+  private static final Pattern REPOSITORY_NAME_PATTERN = Pattern.compile("\\/([a-z0-9._-]+)");
+  private static final String IDENTIFIER_PATTERN = "(:([a-z0-9._-]+)|@(sha256:[0-9a-f]{64}))";
+  private static final Pattern REGION_PATTERN = Pattern.compile("(\\w+-\\w+-\\d+)");
+  private static final Pattern REPOSITORY_URI_PATTERN = Pattern.compile(ACCOUNT_ID_PATTERN.toString() + "\\.dkr\\.ecr\\." +
+    REGION_PATTERN.toString() + ".+" +
+    REPOSITORY_NAME_PATTERN.toString() +
+    IDENTIFIER_PATTERN);
+
+  private final AmazonClientProvider amazonClientProvider;
+
+  private final AccountCredentialsProvider accountCredentialsProvider;
+
+  @Autowired
+  public EcrImageProvider(AmazonClientProvider amazonClientProvider,
+                          AccountCredentialsProvider accountCredentialsProvider) {
+    this.amazonClientProvider = amazonClientProvider;
+    this.accountCredentialsProvider = accountCredentialsProvider;
+  }
+
+  @Override
+  public String getRepositoryName() {
+    return "ECR";
+  }
+
+  @Override
+  public boolean handles(String url) {
+    return isValidEcrUrl(url);
+  }
+
+  @Override
+  public List<EcsDockerImage> findImage(String url) {
+    // HTTP(S) part is not needed.
+    url = url.replace("http://", "").replace("https://", "");
+
+    String accountId = extractAwsAccountId(url);
+    String repository = extractEcrRepositoryName(url);
+    String identifier = extractEcrIdentifier(repository, url);
+    boolean isTag = !(identifier.startsWith("sha256:") && identifier.length() == ("sha256:".length() + 64));
+    String region = extractAwsRegion(url);
+
+    NetflixAmazonCredentials credentials = getCredentials(accountId);
+
+    if (!isValidRegion(credentials, region)) {
+      throw new Error("The repository URI provided does not belong to a region that the credentials have access to or is not a valid region.");
```

Comment:
`s/or is not a valid region/or the region is not valid`.

Reply:
Done.